epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -400,6 +402,37 @@ private void generateClassForModel(ClassToGenerateInfo info) throws IOException
     return constructors;
   }
 
+  private Iterable<MethodSpec> generateMethodsReturningClassType(ClassToGenerateInfo info) {
+    List<MethodSpec> methods = new ArrayList<>(info.getMethodsReturningClassType().size());
+
+    for (MethodInfo methodInfo : info.getMethodsReturningClassType()) {
+      Builder builder = MethodSpec.methodBuilder(methodInfo.name)
+          .addModifiers(methodInfo.modifiers)
+          .addParameters(methodInfo.params)
+          .addAnnotation(Override.class)
+          .returns(info.getGeneratedName());
+
+      StringBuilder statementBuilder = new StringBuilder(String.format("super.%s(",
+          methodInfo.name));
+      boolean first = true;
+      for (ParameterSpec param : methodInfo.params) {
```

Comment:
Ah this is good. I was thinking more of a generic join method like `StringUtils.join`, but your way is better since we have a special usage case. Thanks!